atc/db/resource.go

```diff
@@ -296,22 +296,34 @@ func (r *resource) setResourceConfigScopeInTransaction(tx Tx, scope ResourceConf
 	return nil
 }
 
-func (r *resource) CheckPlan(from atc.Version, interval time.Duration, resourceTypes ResourceTypes, sourceDefaults atc.Source) atc.CheckPlan {
-	return atc.CheckPlan{
-		Name:    r.Name(),
-		Type:    r.Type(),
-		Source:  sourceDefaults.Merge(r.Source()),
-		Tags:    r.Tags(),
-		Timeout: r.CheckTimeout(),
+type ImagePlanner interface {
```

Comment:
I doubt if this is a right place to define this interface. Maybe move to a separate file?